### ISSUE TITLE
Support for modern Spark / Scala versions?

### ISSUE THREAD

**Issue by vvvvalvalval** [gorillalabs/sparkling#76]
AFAICT, it's not possible to use Sparkling with modern Spark versions like `2.4.5`, because they require Scala 2.11 whereas Sparkling requires Scala 2.10 in its interop.

Am I correct in this? If so, is there a plan to upgrade? I'm finding myself more and more limited by having to use Spark `2.1`, all the more so as I add other library dependencies.

**Comment by lins05** [comment]
It works for me with spark 2.4.5. I only used spark core so far and haven't tested spark sql though.

```clojure
  :dependencies [[org.clojure/clojure "1.10.1"]
                 [gorillalabs/sparkling "2.1.3"]]

  :profiles {:provided {:dependencies [[org.apache.spark/spark-core_2.11 "2.4.5"]
                                       [org.apache.spark/spark-sql_2.11 "2.4.5"]
                                       [org.apache.spark/spark-mesos_2.11 "2.4.5"]]}}
```

Also worth to mention `spark-core_2.12` doesn't work, must use `spark-core_2.11`.

**Comment by vvvvalvalval** [comment]
@lins05 thanks, I confirm this works, it appears my incompatibility assumptions were incorrect. I must have only tried incorrect version combinations.

**TL;DR: Sparkling `2.1.3` does work with Spark `2.4.5`, as long as you use the `_2.11` Spark artifacts as explained by @lins05.**
